web/src/app/util/SetFavoriteButton.tsx

```diff
@@ -2,26 +2,35 @@ import React from 'react'
 import IconButton from '@material-ui/core/IconButton'
 import FavoriteFilledIcon from '@material-ui/icons/Star'
 import FavoriteBorderIcon from '@material-ui/icons/StarBorder'
+import Tooltip from '@material-ui/core/Tooltip'
 import Spinner from '../loading/components/Spinner'
 
 interface SetFavoriteButtonProps {
   typeName: 'rotation' | 'service' | 'schedule'
   isFavorite?: boolean
   loading: boolean
   onClick: Function
+
+  tooltipOptions?: {
+    enabled: boolean
+    setMessage?: string // The tooltip displayed when the favorite has already been set
+    unsetMessage?: string // The tooltip displayed when the favorite has not yet been set
```

Comment:
Can we hard-code these and use `typeName`? I feel like any/everywhere we have a `service` fav. button we probably want the same message rather than having the ability to change it.

Reply:
We can, but we made the favorite button generic and this could be useful for the other pages as well (rotations, schedules, etc)

Reply:
It is generic but not type-agnostic. In the case of a rotation or schedule, we'd add a new type and associated messages. We're currently using `typeName` to generate the aria-labels rather than having them passed in which fills a similar role to the tooltip -- both sources of truth should live in the same place I think.

If we rendered a `service` favorite button in two places, we'd have to copy and paste both tooltip messages to both places to get them to match, which makes it seem like it belongs inside the component.